ndcube/extra_coords/lookup_table_coord.py

```diff
@@ -9,318 +9,431 @@
-    physical_types: iterable of `str`, optional
-        The world axis physical types, to be passed to the gWCS frames. This
-        can be used to override the defaults generated by `gwcs`.
-
-    Attributes
-    ----------
-    delayed_models: `list` of `.DelayedLookupTable`
-        A list of all the lookup tables contained by this object.
-    frames: `list` of `gwcs.coordinate_frames.CoordinateFrame`
-        A list of all the gwcs coordinate frame objects corresponding to the
-        lookup tables.
+    if all([u.m.is_equivalent(un) for un in unit]):
+        axes_type = "SPATIAL"
 
-    Notes
-    -----
+    if all([u.pix.is_equivalent(un) for un in unit]):
+        name = "PixelFrame"
+        axes_type = "PIXEL"
+
+    axes_type = tuple([axes_type] * naxes)
 
-    The constructor to this class expects a single coordinate to be passed,
-    i.e. one coordinate frame (e.g. Celestial), although this coordinate could be built
-    from multiple lookup tables.
+    return cf.CoordinateFrame(naxes, axes_type, axes_order, unit=unit,
+                              axes_names=names, name=name, axis_physical_types=physical_types)
 
-    The implementation of this class, however, allows the representation of
-    many coordinate frames in a single instance. This is primarily to allow the
-    ``&`` operator to work between instances of this class, and therefore build
-    up a multi-dimensional WCS object.
 
-    If you wish to build a multi-dimensional `.LookupTableCoord` the correct
-    approach is to construct two instances and then join them into a new
-    combined instance with the ``&`` operator.
+def _generate_tabular(lookup_table, interpolation='linear', points_unit=u.pix, **kwargs):
     """
+    Generate a Tabular model class and instance.
+    """
+    if not isinstance(lookup_table, u.Quantity):
+        raise TypeError("lookup_table must be a Quantity.")  # pragma: no cover
 
-    def __init__(self, *lookup_tables, mesh=True, names=None, physical_types=None):
-        self.delayed_models = []
-        self.frames = []
-
-        if lookup_tables:
-            lt0 = lookup_tables[0]
-            if not all(isinstance(lt, type(lt0)) for lt in lookup_tables):
-                raise TypeError("All lookup tables must be the same type")
-
-            type_map = {
-                u.Quantity: self._from_quantity,
-                Time: self._from_time,
-                SkyCoord: self._from_skycoord
-            }
-            delayed_model, frame = type_map[type(lt0)](lookup_tables,
-                                                       mesh=mesh,
-                                                       names=names,
-                                                       physical_types=physical_types)
-            self.delayed_models = [delayed_model]
-            self.frames = [frame]
+    ndim = lookup_table.ndim
+    TabularND = tabular_model(ndim, name=f"Tabular{ndim}D")
```

Comment:
Is this returning an class or an instance of a class?  If the latter, perhaps it shouldn't be camelcase.

Reply:
It returns a class.